pallets/dapp-staking-v3/src/test/tests.rs

```diff
@@ -2962,3 +2962,135 @@ fn safeguard_on_by_default() {
         assert!(Safeguard::<Test>::get());
     });
 }
+
+#[test]
```

Comment:
It's ok to test all scenarios if you wish, of course, but the setup code bloats 99% of the added test code. Maybe optimize that, and reuse `storage` for creating multiple test externalities.